node/src/blockchain/blockchain_interface/blockchain_interface_web3/lower_level_interface_web3.rs

```diff
@@ -12,13 +12,31 @@ use web3::transports::{Batch, Http};
 use web3::types::{Address, BlockNumber, Filter, Log, TransactionReceipt};
 use web3::{Error, Web3};
 
-#[derive(Debug, PartialEq, Clone)]
-#[allow(clippy::large_enum_variant)]
+#[derive(Debug, PartialEq, Eq, Clone)]
 pub enum TransactionReceiptResult {
     NotPresent,
-    Found(TransactionReceipt),
-    TransactionFailed(TransactionReceipt), // RemoteFailure
-    Error(String),                         // LocalFailure
+    Found(TxReceipt),
+    TransactionFailed(TxReceipt),
+    LocalError(String),
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct TxReceipt {
+    pub transaction_hash: H256,
+    pub block_hash: Option<H256>,
+    pub block_number: Option<U64>,
+    pub status: Option<bool>,
```

Comment:
This is mainly addressed to @utkarshg6. What do you think why couldn't we rely on this assumption?:  

If a transaction hasn't been added onto the blockchain, so it's not part of any block, it seems to me that all the optional fields in the struct will reflect it the same way, they will evaluate necessarily to `None`. Are you sure we cannot do something like: 

```
#[derive(Debug, PartialEq, Eq, Clone)]
pub struct TxReceipt {
    pub transaction_hash: H256,
    pub transaction_block_opt: Option<TransactionBlock>
}

#[derive(Debug, PartialEq, Eq, Clone)]
struct TransactionBlock {
    block_hash_opt: Option<H256>,
    block_number_opt: Option<U64>
}
  
```
I don't know if the responses can contain a divergence in these field but the common sense says that if a transaction doesn't have a block **then it is not confirmed**. That means that we may not need the `status` field at all.  

If you @utkarshg6 think that it's risky depending on that, you still, at minimum, can improve the name of the field as: 

`tx_confirmed_opt` 

where `false` means a failure :man_shrugging:...